modyn/supervisor/internal/eval/strategies/_interval.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from typing import Iterable, Optional
+
+from modyn.config.schema.pipeline.evaluation.strategy._interval import _IntervalEvalStrategyConfig
+from modyn.utils.utils import SECONDS_PER_UNIT
+
+from .abstract import AbstractEvalStrategy
+
+
+class _IntervalEvalStrategy(AbstractEvalStrategy):
+    """See `_IntervalEvalStrategyConfig` for more information."""
+
+    INFINITY = "inf"
+    NEGATIVE_INFINITY = "-inf"
+
+    def __init__(self, config: _IntervalEvalStrategyConfig):
+        super().__init__(config)
+
+    def get_eval_intervals(
+        self, first_timestamp: int, last_timestamp: int
```

Comment:
My only question is about the interface here. We mentioned before that sometimes, in after_training evaluation, we don't have the full information (i.e., we don't know when the next trigger will be). What I don't see yet is how the different strategies will differ in their interface depending on the available information. We have the common first_timestamp, last_timestamp interface, which gives you the first and last sample in the current trigger. But how do you pass additional information to eval strategies, if you have it?

Reply:
previously I implemented it using the `EvalTrigger` concept that we now removed with the new abstraction. I haven't implemented the adjusted version yet, but I'll do it either with additional (optional) parameters to `get_eval_intervals`, a separate member function call that adds state to the stategies, or handle some strategies explicitly in the pipeline executor (e.g. the `BetweenTriggerEvalStrategy` is quite different). For simplicity, it might make sense to handle that in the pipeline executor.